eras/babbage/impl/src/Cardano/Ledger/Babbage/TxBody.hs

```diff
@@ -0,0 +1,897 @@
+                  mintX
+                  scriptIntegrityHashX
+                  adHashX
+                  txnetworkidX
+            )
+
+{-# COMPLETE TxBody #-}
+
+instance (c ~ Crypto era) => HashAnnotated (TxBody era) EraIndependentTxBody c
+
+-- ==============================================================================
+-- We define these accessor functions manually, because if we define them using
+-- the record syntax in the TxBody pattern, they inherit the (BabbageBody era)
+-- constraint as a precondition. This is unnecessary, as one can see below
+-- they need not be constrained at all. This should be fixed in the GHC compiler.
+
+inputs' :: TxBody era -> Set (TxIn (Crypto era))
+collateral' :: TxBody era -> Set (TxIn (Crypto era))
+referenceInputs' :: TxBody era -> Set (TxIn (Crypto era))
+outputs' :: TxBody era -> StrictSeq (TxOut era)
+collateralReturn' :: TxBody era -> StrictMaybe (TxOut era)
+totalCollateral' :: TxBody era -> Coin
+certs' :: TxBody era -> StrictSeq (DCert (Crypto era))
+txfee' :: TxBody era -> Coin
+wdrls' :: TxBody era -> Wdrl (Crypto era)
+vldt' :: TxBody era -> ValidityInterval
+update' :: TxBody era -> StrictMaybe (Update era)
+reqSignerHashes' :: TxBody era -> Set (KeyHash 'Witness (Crypto era))
+adHash' :: TxBody era -> StrictMaybe (AuxiliaryDataHash (Crypto era))
+mint' :: TxBody era -> Value (Crypto era)
+scriptIntegrityHash' :: TxBody era -> StrictMaybe (ScriptIntegrityHash (Crypto era))
+inputs' (TxBodyConstr (Memo raw _)) = _inputs raw
+
+txnetworkid' :: TxBody era -> StrictMaybe Network
+
+collateral' (TxBodyConstr (Memo raw _)) = _collateral raw
+
+referenceInputs' (TxBodyConstr (Memo raw _)) = _referenceInputs raw
+
+outputs' (TxBodyConstr (Memo raw _)) = _outputs raw
+
+collateralReturn' (TxBodyConstr (Memo raw _)) = _collateralReturn raw
+
+totalCollateral' (TxBodyConstr (Memo raw _)) = _totalCollateral raw
+
+certs' (TxBodyConstr (Memo raw _)) = _certs raw
+
+wdrls' (TxBodyConstr (Memo raw _)) = _wdrls raw
+
+txfee' (TxBodyConstr (Memo raw _)) = _txfee raw
+
+vldt' (TxBodyConstr (Memo raw _)) = _vldt raw
+
+update' (TxBodyConstr (Memo raw _)) = _update raw
+
+reqSignerHashes' (TxBodyConstr (Memo raw _)) = _reqSignerHashes raw
+
+adHash' (TxBodyConstr (Memo raw _)) = _adHash raw
+
+mint' (TxBodyConstr (Memo raw _)) = _mint raw
+
+scriptIntegrityHash' (TxBodyConstr (Memo raw _)) = _scriptIntegrityHash raw
+
+txnetworkid' (TxBodyConstr (Memo raw _)) = _txnetworkid raw
+
+--------------------------------------------------------------------------------
+-- Serialisation
+--------------------------------------------------------------------------------
+
+instance
+  ( Era era,
+    Compactible (Core.Value era)
+  ) =>
+  ToCBOR (TxOut era)
+  where
+  toCBOR (TxOutCompact addr cv) =
+    encodeListLen 2
+      <> toCBOR addr
+      <> toCBOR cv
+  toCBOR (TxOutCompactDatum addr cv d) =
+    encodeListLen 4
+      <> toCBOR True
+      <> toCBOR addr
+      <> toCBOR cv
+      <> toCBOR d
```

Comment:
this feels kinda hacky, but putting a break before the `encodeListLen` makes tests fail